src/leap/common/events/signals.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# proto.py
+# Copyright (C) 2015 LEAP
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful",
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+"""
+Communication protocol definitions.
+"""
+
+
+from enum import Enum
+
+
+signals = Enum(
+    "CLIENT_SESSION_ID",
+    "CLIENT_UID",
+    "SOLEDAD_CREATING_KEYS",
+    "SOLEDAD_DONE_CREATING_KEYS",
+    "SOLEDAD_UPLOADING_KEYS",
+    "SOLEDAD_DONE_UPLOADING_KEYS",
+    "SOLEDAD_DOWNLOADING_KEYS",
+    "SOLEDAD_DONE_DOWNLOADING_KEYS",
+    "SOLEDAD_NEW_DATA_TO_SYNC",
+    "SOLEDAD_DONE_DATA_SYNC",
+    "UPDATER_NEW_UPDATES",
+    "UPDATER_DONE_UPDATING",
+    "RAISE_WINDOW",
+    "SMTP_SERVICE_STARTED",
+    "SMTP_SERVICE_FAILED_TO_START",
+    "SMTP_RECIPIENT_ACCEPTED_ENCRYPTED",
+    "SMTP_RECIPIENT_ACCEPTED_UNENCRYPTED",
+    "SMTP_RECIPIENT_REJECTED",
+    "SMTP_START_ENCRYPT_AND_SIGN",
+    "SMTP_END_ENCRYPT_AND_SIGN",
+    "SMTP_START_SIGN",
+    "SMTP_END_SIGN",
+    "SMTP_SEND_MESSAGE_START",
+    "SMTP_SEND_MESSAGE_SUCCESS",
+    "SMTP_SEND_MESSAGE_ERROR",
+    "SMTP_CONNECTION_LOST",
+    "IMAP_SERVICE_STARTED",
+    "IMAP_SERVICE_FAILED_TO_START",
+    "IMAP_CLIENT_LOGIN",
+    "IMAP_FETCHED_INCOMING",
+    "IMAP_MSG_PROCESSING",
+    "IMAP_MSG_DECRYPTED",
+    "IMAP_MSG_SAVED_LOCALLY",
+    "IMAP_MSG_DELETED_INCOMING",
+    "IMAP_UNHANDLED_ERROR",
+    "IMAP_UNREAD_MAIL",
```

Comment:
What do you mean by (str, int) ? Would these be arguments to the event? Currently, I added a simple "content" string, but I will add some kind of serialization to allow for *contents instead.